### PR TITLE
[ios] Remove invisible taxi button from route preview

### PR DESCRIPTION
**Pull request by nojus297**
As I understand, taxi routing was remove a while ago, but invisible taxi icon remained in route preview. Deleted it; icons are now nicely center aligned.

**Was**
IMG_0580
**After this fix**
IMG_0581

### CODE REVIEW

**Comment by biodranik** [comment]
Thanks! Can you please replace the taxi icon with a helicopter icon (and leave it hidden for now)? We plan to add a "helicopter" routing :)

**Comment by nojus297** [comment]
Took a while to confirm the helicopter wasn't a joke:)

Added back the button to storyboard and unmarked "Installed" to hide it. It does not have an icon and is not connected to anything else.

I did not "Uninstall" the constraints though. This gives out a few warnings, but it will require less clicks to re-enable the helicopter button. Is that OK?

**Review by biodranik — COMMENTED** [review]
Thanks! What kind of warnings did you get?

Can you please check how this icon will look like (and share a screenshot)? https://github.com/mapbox/maki/blob/main/icons/heliport.svg

If it's ok, you can add it (and probably update our data/copyright.html with Maki license).

**Comment by nojus297** [comment]
Below are the image of how the helicopter looks. Also while there, I tried out the bicycle icon from the same repo. IMO it looks better
IMG_0609
in comparison with the old bicycle icon:
IMG_0581

**Comment by nojus297** [comment]
The warnings given are multiple of `Constraint referencing items turned off in current configuration. Turn off this constraint in the current configuration`. These can be turned of just by uninstalling the constraints too (not just the button).

**Comment by endim8** [comment]
i like the new bicycle icon, i'll make a PR adding these things to android as well 👍

**Comment by biodranik** [comment]
Thanks! The icons look good. Maybe the maki icons size can be slightly increased somehow, to (almost) match other icons height? The same "resized" icon can be reused on all platforms.

Also please don't forget about the copyright.

**Comment by biodranik** [comment]
@nojus297 to confirm: in this PR helicopter is invisible, right?

**Comment by nojus297** [comment]
> @nojus297 to confirm: in this PR helicopter is invisible, right?

yes, invisible and not taking space (allowing other icons to be centered).

**Comment by nojus297** [comment]
> Thanks! The icons look good. Maybe the maki icons size can be slightly increased somehow, to (almost) match other icons height? The same "resized" icon can be reused on all platforms.

I scaled both new icons by 220% (with inkscape). Do they need to be even bigger? (Note that the new bike is bigger than the old one, and both of the new icons are closer to the edges than ones of car and pedestrian).

Also the new icons are more horizontal, so scaling them to match height might make them very wide.

**Comment by biodranik** [comment]
The idea is to make them slightly more recognizable to people with bad eyesight. Let's compare the look of the current version with a slight increase.

**Comment by nojus297** [comment]
Sharing some screenshots to compare:
previous (220% from maki)
IMG_0611
scaled by 5% (231% total)
IMG_0613
scaled by 10% (241% total)
IMG_0614

**Comment by biodranik** [comment]
The difference is almost negligible, but bigger looks better. What do you think?

**Comment by nojus297** [comment]
I prefer +5%, I'm okay with +10% but definitely would not want to go any bigger, already looks a bit cramped to my eyes. 

However, if we go with +10%, we should probably scale the car as well, as it looks a bit small in comparison (when not selected)
image

**Comment by biodranik** [comment]
Good point. Ok, 5% is also good. Let's move on with implementing the helicopter routing in the code in another PR :) @nojus297 can you start to draft it? We will help by adding support in the core.

**Comment by nojus297** [comment]
Added the new icons with the license. Also some Helicopter enum values, guessing best switch case in some places. (Helicopter is currently not visible).

File: iphone/Maps/Classes/CustomViews/NavigationDashboard/Views/RoutePreview/MWMRoutePreview.mm [hunk 1]
```diff
@@ -106,7 +107,7 @@ - (void)statePrepare
     progress.second.state = MWMCircularProgressStateNormal;
 
   //if (!MWMLocationManager.lastLocation || !Platform::IsConnected())
-    [self.taxi removeFromSuperview];
+    //[self.taxi removeFromSuperview];
```
Comment (biodranik) [inline iphone/Maps/Classes/CustomViews/NavigationDashboard/Views/RoutePreview/MWMRoutePreview.mm:110]: These comments can be removed.

File: iphone/Maps/Core/Location/MWMLocationManager.mm [hunk 2]
```diff
@@ -319,6 +319,7 @@ + (void)setMyPositionMode:(MWMMyPositionMode)mode
     {
     case MWMRouterTypeVehicle: manager.geoMode = GeoMode::VehicleRouting; break;
     case MWMRouterTypePublicTransport:
+    case MWMRouterTypeHelicopter:
```
Comment (biodranik) [inline iphone/Maps/Core/Location/MWMLocationManager.mm:322]: No, it's better to insert it at the end. Please try to understand what the previous code did and decide if it is relevant here.

Comment (biodranik) [reply iphone/Maps/Core/Location/MWMLocationManager.mm:322]: Also, sometimes it is better to leave a warning and come back to fix it later.

**Review by biodranik — COMMENTED** [review]
Good start! Let's check how other routers are working and try to do the same. The difference will be is that the route can be calculated instantly via specified points.

File: iphone/Maps/Core/Location/MWMLocationManager.mm (hunk 2 above)
Comment (nojus297) [reply iphone/Maps/Core/Location/MWMLocationManager.mm]: Is this ok now?

**Comment by nojus297** [comment]
Are we planning to implement the actual routing in this PR too? Could we merge this PR now and add routing in separate PR, leaving this one just for icons?

**Comment by biodranik** [comment]
It is a good idea. But then let's create a new PR and move there these enums and switch cases. There should be no any side effect in the master.

**Comment by nojus297** [comment]
Removed all the helicopters from routing logic.

**Review by biodranik — APPROVED** [review]
Thanks! Can be merged if all checks pass.

**Comment by nojus297** [comment]
Looks like it passed

**Comment by biodranik** [comment]
Yes, if you plan to implement it and actually fill these enum switch cases with a valid code )

**Pull request merged**